### PR TITLE
[TD-2623] Resolve conversation on Chat-widget and Dashboard once the ticket/chat gets closed on Zendesk.

### PR DESCRIPTION
**Pull request by raj5036**
### What do you want to achieve?
- Resolve conversation on Chat-widget and Dashboard once the ticket/chat gets closed on Zendesk.

### PR Checklist

- [X] I have tested it locally and all functionalities are working fine.

### How was the code tested?

- Tested locally. All functionality is working fine.
image
image
image

### What new thing you came across while writing this code? 
- How we resolve conversation on chat-widget side.

NOTE: Make sure you're comparing your branch with the correct base branch

### CODE REVIEW

File: webplugin/js/app/zendesk-chat-service.js [hunk 1]
```diff
@@ -74,6 +74,25 @@ function ZendeskChatService() {
...
+                    //Call API to resolve the chat on Dashboard
+                    kommunicate.client.resolveConversation({ 
+                        groupId: CURRENT_GROUP_DATA.tabId 
+                    }, function(err, result) {
+                        if (err || !result) {
+                            console.log('An error occurred while resolving conversation ',err);
+                            return;
+                        }
+                        console.log(result)
+                    });
```
Comment (gargshantnu) [inline webplugin/js/app/zendesk-chat-service.js:94]: can you move this whole logic to new function?

File: webplugin/js/app/zendesk-chat-service.js [hunk 2]
```diff
@@ -74,6 +74,25 @@ function ZendeskChatService() {
                     _this.handleZendeskAgentMessageEvent(eventDetails);
                 } else if (eventDetails.type == "chat.file") { //If agent sends file attachments
                     _this.handleZendeskAgentFileSendEvent(eventDetails);
+                } else if (eventDetails.type == "chat.memberleave") {
+                    //Resolve conversation on widget
+                    KommunicateConstants.CONVERSATION_RESOLVED_STATUS = true;
```
Comment (gargshantnu) [inline webplugin/js/app/zendesk-chat-service.js:79]: i guess this is a constant. are we supposed to change constants as well?

Comment (raj5036) [reply webplugin/js/app/zendesk-chat-service.js]: No it wasn't necessary. Removed it.

File: webplugin/js/app/zendesk-chat-service.js (hunk 1 above)
Comment (raj5036) [reply webplugin/js/app/zendesk-chat-service.js]: Done.

File: webplugin/js/app/zendesk-chat-service.js [hunk 3]
```diff
@@ -247,7 +249,27 @@ function ZendeskChatService() {
...
+        //Resolve conversation on widget
+        KommunicateUI.showClosedConversationBanner(
+            true
+        );
+        KommunicateUI.isConvJustResolved = true;
+        
+        //Call API to resolve the conversation on Dashboard
+        kommunicate.client.resolveConversation({ 
+            groupId: CURRENT_GROUP_DATA.tabId 
+        }, function(err, result) {
```
Comment (gargshantnu) [inline webplugin/js/app/zendesk-chat-service.js:264]: i guess we are supposed to show some new text "Start Chat Again", and on click of that it should create a new conversation. can you check once with Shweta to verify if current functionality is fine or not.

Comment (raj5036) [reply webplugin/js/app/zendesk-chat-service.js:264]: Few changes were required. Added them.

**Review by gargshantnu — DISMISSED** [review]

**Review by Prochnost55 — APPROVED** [review]

**Pull request merged**